hoomd/mpcd/CollisionMethod.cuh

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2009-2025 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/CollisionMethod.cuh
+ * \brief Declaration of CUDA kernels for mpcd::CollisionMethod
+ */
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/Index1D.h"
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+namespace kernel
+    {
+
+void gpu_check_collision_warnings(Scalar4 d_initial_velo,
+                                  const Scalar4* d_vel_embed,
+                                  const unsigned int* d_embed_group,
+                                  const unsigned int* d_body,
+                                  const unsigned int* d_rtag,
+                                  const uint64_t timestep);
+
+void gpu_store_initial_embedded_group_velocities(Scalar4 d_initial_velo,
+                                                 const Scalar4* d_vel_embed,
+                                                 const unsigned int* d_embed_group,
+                                                 const uint64_t timestep);
```

Comment:
These driver functions should only take the arguments they need. In this case, I think you won't need the `timestep`. However, you will need the number of particles in the group and also a tunable kernel block size. The functions should also usually return a `cudaError_t` (error code), which will end up being `cudaSuccess`.
```suggestion
cudaError_t store_initial_embedded_group_velocities(Scalar4 d_initial_velo,
                                                 const Scalar4* d_vel_embed,
                                                 const unsigned int* d_embed_group,
                                                 const unsigned int num_group,
                                                 const unsigned int block_size);
```